library/std/src/net/ip.rs

```diff
@@ -1042,6 +1042,20 @@ impl From<[u8; 4]> for IpAddr {
     }
 }
 
+// One shouldn't use this union directly but please use
+// `Ipv6Addr::segments` or `Ipv6Addr::octets`.
+//
+// # Safety
+//
+// This union assumes that all fields are interpreted as big endian.
+// Also, all of its field has the same size.
+#[repr(C)]
+union Ipv6Bytes {
```

Comment:
I don't quite understand why do we need to declare this union ourselves? Why not just use the bytes directory rather than going through a layer of union like `[a.to_be(), b.to_be(), ...]`?

Reply:
I don't quite get your point. What is bytes directory? Is the old code (in master)  using bytes directory approach?

Reply:
Oh, I mean as in "use the bytes directly", sorry typo.

Reply:
Ah, thanks for explaining. Here the codegen diff: https://rust.godbolt.org/z/cE9zqd

Reply:
Oh, the binary looks much shorter. Looks like performance is the only reason, should we explain that union is used for performance reasons here?

Reply:
Good point. Will do.